### DISCUSSION TITLE
Question: cold start vs warm start compile times

### DISCUSSION THREAD

**Discussion by sambacha** [foundry-rs/foundry#521] [Q&A] [answered]
Basically wondering how much of a difference (if any statistically measurable amount exists that is) would there be if forge compiles from a dead cold start vs a pre-warmed state. 

Pre warming would be compiling something like

```sol
pragma solidity ^0.8.0;contract dummy{}"
```

for `x` amount of times. 

Is there a potential worthwhile gain to warrant exploring this? 

Cheers

**Comment by gakonst** [comment, marked as answer]
WDYM pre-warmed state? There's caching in the compiler already, which saves some time. Calling it N times doesn't really do much. We also want to add forking caching to disk when pinned against a block, to avoid re-making network requests in forked tests.

**Comment by sambacha** [comment]
> WDYM pre-warmed state? There's caching in the compiler already, which saves some time. Calling it N times doesn't really do much. We also want to add forking caching to disk when pinned against a block, to avoid re-making network requests in forked tests.

So forge will support using an external store for cache persistence, e.g. redis (if you wanted to)?

**Comment by onbjerg** [comment]
No, just disk cache :slightly_smiling_face:

**Comment by gakonst** [comment]
Could do anything really but let's just go with disk cache, keyed by pinned block like hardhat
